client/asset/dcr/dcr.go

```diff
@@ -935,33 +935,101 @@ func (dcr *ExchangeWallet) findExistingAddressBasedTxHistoryDB() (string, error)
 	return "", nil
 }
 
-func (dcr *ExchangeWallet) startTxHistoryDB(ctx context.Context) (*dex.ConnectionMaster, error) {
-	var dbPath string
-	if spvWallet, ok := dcr.wallet.(*spvWallet); ok {
-		initialAddress, err := spvWallet.InitialAddress(ctx)
+// rpcTxHistoryDBInfo is the information stored in the rpc-tx-dbs.json file.
+// Each tx history db file is associated with a set of accounts used to
+// configure an rpc wallet.
+type rpcTxHistoryDBInfo struct {
+	Addresses *rpcWalletAddresses `json:"addresses"`
+	FileName  string              `json:"fileName"`
+}
+
+func (dcr *ExchangeWallet) rpcHistoryDBInfosFilePath() string {
+	return filepath.Join(dcr.walletDir, "rpc-tx-dbs.json")
+}
+
+func (dcr *ExchangeWallet) existingRPCTxHistoryDBs() ([]*rpcTxHistoryDBInfo, error) {
+	if b, err := os.ReadFile(dcr.rpcHistoryDBInfosFilePath()); err == nil {
+		var dbInfos []*rpcTxHistoryDBInfo
+		err = json.Unmarshal(b, &dbInfos)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("unable to unmarshal rpc tx dbs file: %v", err)
 		}
+		return dbInfos, nil
+	} else if !errors.Is(err, os.ErrNotExist) {
+		return nil, fmt.Errorf("unable to rpc tx dbs file: %v", err)
+	}
+
+	return []*rpcTxHistoryDBInfo{}, nil
+}
+
+func (dcr *ExchangeWallet) updateRPCTxHistoryDBs(dbInfos []*rpcTxHistoryDBInfo) error {
+	b, err := json.Marshal(dbInfos)
+	if err != nil {
+		return fmt.Errorf("unable to marshal rpc tx dbs: %v", err)
+	}
+
+	err = os.WriteFile(dcr.rpcHistoryDBInfosFilePath(), b, 0644)
+	if err != nil {
+		return fmt.Errorf("unable to write rpc tx dbs file: %v", err)
+	}
+
+	return nil
+}
 
-		dbPath = dcr.txHistoryDBPath(initialAddress)
+func (dcr *ExchangeWallet) findRPCTxHistoryDBFile(ctx context.Context, wallet *rpcWallet) (string, error) {
+	dbInfos, err := dcr.existingRPCTxHistoryDBs()
+	if err != nil {
+		return "", err
 	}
 
-	if dbPath == "" {
-		addressPath, err := dcr.findExistingAddressBasedTxHistoryDB()
+	for _, dbInfo := range dbInfos {
+		match, err := wallet.addressesMatchAccounts(ctx, dbInfo.Addresses)
```

Comment:
Why do we need different DBs for different account configurations? It's not a new wallet.

Reply:
You need all the accounts to be the same to have the same tx history, not just one. If they change the mixed account, but not the unmixed account, we cannot show the original history.